eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,93 @@
+$Language = "rust"
+$LanguageDisplayName = "Rust"
+$PackageRepository = "crates.io"
+$packagePattern = "Cargo.toml"
+$MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/releases/latest/rust-packages.csv"
+$GithubUri = "https://github.com/Azure/azure-sdk-for-rust"
+$PackageRepositoryUri = "https://crates.io/crates"
+
+function Get-AllPackageInfoFromRepo ([string] $ServiceDirectory) {
+    $allPackageProps = @()
+    Push-Location $RepoRoot
+    try {
+        $searchPath = Join-Path $RepoRoot 'sdk' -Resolve
+
+        if ($ServiceDirectory -and $ServiceDirectory -ne 'auto') {
+            $searchPath = Join-Path 'sdk' $ServiceDirectory
+        }
+
+        $packages = cargo metadata --format-version 1
+        | ConvertFrom-Json -AsHashtable
+        | Select-Object -ExpandProperty packages
+        | Where-Object { $_.manifest_path.StartsWith($searchPath) }
+
+        $packageManifests = @{}
+        foreach ($package in $packages) {
+            if ($package.manifest_path -replace '\\', '/' -match '/sdk/([^/]+)/') {
+                $package.ServiceDirectoryName = $Matches[1]
+            }
+            else {
+                # ignore manifests that are not in a service directory
+                continue
+            }
+
+            $package.RelativePath = (Split-Path $package.manifest_path -Parent).Replace($RepoRoot, "").SubString(1)
```

Comment:
Why the Substring(1)? What is the first char?

Reply:
it's a path separator

`"/mnt/vss/_work/1/s/sdk/identity/azure_identity".Replace("/mnt/vss/_work/1/s", "")` => `"/sdk/identity/azure_identity"`

We could also use `.TrimStart("\/")`